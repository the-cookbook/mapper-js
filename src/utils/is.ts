export default {
  nullOrUndefined: (value: unknown): boolean => value === null || value === undefined,
  object: <T extends Record<string, unknown>>(value: unknown): value is T =>
    value !== null && typeof value === 'object' && Object.prototype.toString.call(value) === '[object Object]',
  array: <T = unknown>(value: unknown): value is T[] => Array.isArray(value),
};
