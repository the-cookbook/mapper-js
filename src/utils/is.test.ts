import is from './is';

describe('utils/is', () => {
  it('should match snapshot', () => {
    expect(is).toMatchSnapshot();
  });

  describe('is.nullOrUndefined()', () => {
    it('should flag data type correctly', () => {
      expect(is.nullOrUndefined(null)).toBe(true);
      expect(is.nullOrUndefined(undefined)).toBe(true);
      expect(is.nullOrUndefined('')).toBe(false);
      expect(is.nullOrUndefined('123')).toBe(false);
    });
  });

  describe('is.object()', () => {
    it(`should flag element data type correctly`, () => {
      expect(is.object({})).toBe(true);
      expect(is.object([])).toBe(false);
      expect(is.object(1)).toBe(false);
      expect(is.object(true)).toBe(false);
      expect(is.object(null)).toBe(false);
      expect(is.object('cd ..')).toBe(false);
    });
  });

  describe('is.array()', () => {
    it(`should flag element data type correctly`, () => {
      expect(is.array([])).toBe(true);
      expect(is.array({})).toBe(false);
      expect(is.array(1)).toBe(false);
      expect(is.array(true)).toBe(false);
      expect(is.array(null)).toBe(false);
      expect(is.array('cd ..')).toBe(false);
    });
  });
});
