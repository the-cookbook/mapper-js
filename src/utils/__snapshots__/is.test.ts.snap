// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/is should match snapshot 1`] = `
Object {
  "array": [Function],
  "nullOrUndefined": [Function],
  "object": [Function],
}
`;
